crypto/rand/rand_win.c

```diff
@@ -704,14 +704,13 @@ static void readscreen(void)
     hBitmap = CreateCompatibleBitmap(hScrDC, w, n);
 
     /* Get bitmap properties */
-    GetObject(hBitmap, sizeof(BITMAP), (LPSTR) & bm);
-    size = (unsigned int)bm.bmWidthBytes * bm.bmHeight * bm.bmPlanes;
-
-    bi.biSize = sizeof(BITMAPINFOHEADER);
+    GetObject(hBitmap, sizeof(bm), (LPSTR)&bm);
+    size = (unsigned int)4 * bm.bmHeight * bm.bmPlanes;
+    bi.biSize = sizeof(bi);
     bi.biWidth = bm.bmWidth;
     bi.biHeight = bm.bmHeight;
     bi.biPlanes = bm.bmPlanes;
-    bi.biBitCount = bm.bmBitsPixel;
+    bi.biBitCount = 32;
```

Comment:
So that size should be 4 \* bm.bmWidth \* bm.bmHeight \* bm.bmPlanes. Though I don't quite understand what are Planes, so one might as well force it to 1, just like with biBitCount. In which case size would be 4 \* bm.bmWidth \* bm.bmHeight.

Reply:
Additional commit pushed